File: my-app/src/types/password-reset.ts
export interface PasswordResetRequestItem {
  id: string;
  user_id: string;
  username: string;
  display_name: string | null;
  created_at: string;
}

export interface PasswordResetListResponse {
  items: PasswordResetRequestItem[];
}

export interface PasswordResetCountResponse {
  count: number;
}

export interface PasswordResetApproveResponse {
  approved_count: number;
}

File: backend/schemas/password_reset.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class ForgotPasswordRequest(BaseModel):
    username: str


class ForgotPasswordResponse(BaseModel):
    message: str


class PasswordResetRequestItem(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    username: str
    display_name: str | None
    created_at: datetime


class PasswordResetListResponse(BaseModel):
    items: list[PasswordResetRequestItem]


class PasswordResetCountResponse(BaseModel):
    count: int


class PasswordResetApproveRequest(BaseModel):
    request_ids: list[uuid.UUID]


class PasswordResetApproveResponse(BaseModel):
    approved_count: int
